camel-core/src/main/java/org/apache/camel/cms/orchestrator/aggregator/CamelPayloadAggregator.java

```diff
@@ -57,8 +61,32 @@ public String getId() {
     return aggregator.getId();
   }
 
-  public static Payload getPayload(byte[] bytes)
+  public Payload getPayload(byte[] bytes)
       throws IOException, ClassNotFoundException {
-    return ByteUtils.fromBytes(bytes, Payload.class);
+    Payload payload = typeConverterRegistry.lookup(Payload.class, byte[].class).convertTo(Payload.class, bytes);
+    Class bodyType = payload.getBodyType();
+    if (bodyType == null) {
+      return null;
```

Comment:
if bodyType is null Payload must not be null right? inside payload value should be null